python/paddle/fluid/layers/collective.py

```diff
@@ -134,10 +134,6 @@ def _c_reducescatter(x, nranks, ring_id=0, use_calc_stream=False):
     if not isinstance(x, Variable):
         raise TypeError('x must be a Variable')
 
-    if x.shape[0] % nranks != 0:
```

Comment:
keep this check when shape[0] larger than 0

Reply:
done.